RummikubGameApp/RummikubGameApp/GameLogic.cs

```diff
@@ -0,0 +1,340 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Drawing;
+using System.Windows.Forms;
+using System.Text.RegularExpressions;
+
+namespace RummikubGameApp
+{
+    public class GameLogic
+    {
+        private Tile[] tiles;
+        public static Tile[] hand;
+        private Tile[] AIhand;
+        private Tile[,] deck;
+        private int currentPlayerIndex;
+        private int tilesIndex;
+
+        public GameLogic()
+        {
+            this.tiles = new Tile[106];
+            hand = new Tile[106];
+            this.AIhand = new Tile[106];
+            this.deck = new Tile[4, 20];
+            this.currentPlayerIndex = 2;
+            this.currentPlayerIndex = 28;
+        }
+
+        public Tile[] StartGame()
+        {
+            // Initialize the game by creating the tiles and shuffling them
+            
+            InitializeTiles();
+            Shuffle();
+            Tile[] arr = Deal();
+            // Set the first player as the current player
+            currentPlayerIndex = 0;
+            return arr;
+        }
+
+
+        public void InitializeTiles()
+        {
+            int index = 0;
+            for (int i = 1; i <= 13; i++)
+            {
+                tiles[index++] = new Tile { Color = "Red", Value = i };
+                tiles[index++] = new Tile { Color = "Red", Value = i };
+                tiles[index++] = new Tile { Color = "Blue", Value = i };
+                tiles[index++] = new Tile { Color = "Blue", Value = i };
+                tiles[index++] = new Tile { Color = "Yellow", Value = i };
+                tiles[index++] = new Tile { Color = "Yellow", Value = i };
+                tiles[index++] = new Tile { Color = "Black", Value = i };
+                tiles[index++] = new Tile { Color = "Black", Value = i };
+            }
+            tiles[index++] = new Tile { Color = "Joker", Value = 0 };
+            tiles[index++] = new Tile { Color = "Joker", Value = 14 };
+        }
+        public Tile[] Shuffle()
+        {
+            // Shuffle the tiles randomly
+            Random random = new Random();
+            for (int i = 0; i < tiles.Length; i++)

```

Comment:
Shuffle can stop at tiles.Length -1 since there is no need to shuffle the last one (it does nothing because you random from i to length)